### PR TITLE
fix(es/parser): Handle `async` in `for...of`

### PR DESCRIPTION
**Pull request by magic-akari**
Invalid:

```JavaScript
for (async of []) {}
```

Valid

```JavaScript
 for await (async of []) {}
```

### CODE REVIEW

**Comment by magic-akari** [comment]
I am thinking about `for( async of =>`

File: crates/swc_ecma_parser/src/parser/stmt.rs [hunk 1]
```diff
@@ -1173,6 +1173,20 @@ impl<'a, I: Tokens> Parser<I> {
 
         let init = if eat_exact!(self, ';') {
             return self.parse_normal_for_head(None);
+        } else if is!(self, "async") && peeked_is!(self, "of") {
```
Comment (magic-akari) [inline crates/swc_ecma_parser/src/parser/stmt.rs:1176]: I am trying to look forward two words. Is there any way to do this? @kdy1

Comment (kdy1) [reply crates/swc_ecma_parser/src/parser/stmt.rs:1176]: No, it's not possible at now

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_parser

**Pull request merged**